homeassistant/components/utility_meter/sensor.py

```diff
@@ -223,7 +225,49 @@ async def async_setup_platform(
     )
 
 
-class UtilityMeterSensor(RestoreEntity, SensorEntity):
+@dataclass
+class UtilitySensorExtraStoredData(SensorExtraStoredData):
+    """Object to hold extra stored data."""
+
+    last_period: Decimal
+    last_reset: datetime | None
+    status: str
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the utility sensor data."""
+        data = super().as_dict()
+        data["last_period"] = str(self.last_period)
+        if isinstance(self.last_reset, (datetime)):
+            data["last_reset"] = self.last_reset.isoformat()
+        data["status"] = self.status
+
+        return data
+
+    @classmethod
+    def from_dict(cls, restored: dict[str, Any]) -> UtilitySensorExtraStoredData | None:
+        """Initialize a stored sensor state from a dict."""
+        extra = SensorExtraStoredData.from_dict(restored)
+        if extra is None:
+            return None
+
+        try:
+            last_period: Decimal = Decimal(restored["last_period"])
```

Comment:
This will throw `decimal.InvalidOperation` if the stored data is invalid, let's catch that and return `None` if caught.

Reply:
fixed